### PR TITLE
ELEMENTS-861: validate if filename is eligible to check for extension

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

File: widgets/nuxeo-uploader-behavior.html [hunk 1]
```diff
@@ -124,8 +124,8 @@
      * @return {Boolean} true if the current file is accepted by the current provider, false otherwise
      * */
     DefaultUploadProvider.prototype.accepts = function (file) {
-      const mimeType = ((file.type !== '') ? file.type.match(/^[^/]*\//)[0] : null);
-      const fileType = file.name.match(/\.[^.]*$/)[0];
+      const mimeType = (file.type !== '') ? file.type.match(/^[^/]*\//)[0] : null;
+      const fileType = (file.name.indexOf('.') !== -1) ? file.name.match(/\.[^.]*$/)[0] : null;
```
[the remark this answers is not in the document]
Comment (nmpcunha) [reply widgets/nuxeo-uploader-behavior.html]: Indeed! Refactoring that small piece of code 👍

File: widgets/nuxeo-uploader-behavior.html [hunk 2]
```diff
@@ -124,10 +124,12 @@
      * @return {Boolean} true if the current file is accepted by the current provider, false otherwise
      * */
     DefaultUploadProvider.prototype.accepts = function (file) {
-      const mimeType = ((file.type !== '') ? file.type.match(/^[^/]*\//)[0] : null);
-      const fileType = file.name.match(/\.[^.]*$/)[0];
-
-      return !(this.accept && !(this.accept.indexOf(mimeType) > -1 || this.accept.indexOf(fileType) > -1));
+      if (!this.accept) {
+        return false;
```
Comment (guirenard) [inline widgets/nuxeo-uploader-behavior.html]: if `this.accept` is empty then we should accept any type of file and therefore return `true` !

**Comment by guirenard** [comment]
:+1:

File: widgets/nuxeo-uploader-behavior.html [hunk 3]
```diff
@@ -124,10 +124,12 @@
...
      * */
     DefaultUploadProvider.prototype.accepts = function (file) {
-      const mimeType = ((file.type !== '') ? file.type.match(/^[^/]*\//)[0] : null);
-      const fileType = file.name.match(/\.[^.]*$/)[0];
-
-      return !(this.accept && !(this.accept.indexOf(mimeType) > -1 || this.accept.indexOf(fileType) > -1));
+      if (!this.accept) {
+        return true;
+      }
+      const mimeType = (file.type !== '') ? file.type.match(/^[^/]*\//)[0] : null;
```
Comment (Gabez0r) [inline widgets/nuxeo-uploader-behavior.html]: nit: could be made shorter
```suggestion
      const mimeType = !!file.type && file.type.match(/^[^/]*\//)[0];
```

**Comment by Gabez0r** [comment]
👍

**Pull request merged**